source/HeadsUpDisplay.hx

```diff
@@ -0,0 +1,136 @@
+package;
+
+/**
+ * ...
+ * @author Dillon Woollums
+ */
+import flixel.group.FlxGroup;
+import flixel.FlxSprite;
+import flixel.text.FlxText;
+import flixel.util.FlxColor;
+
+class HeadsUpDisplay extends FlxGroup
+{
+	private var scoreLabel:FlxText;
+	private var timeLabel:FlxText;
+	private var coinIcon:FlxSprite;
+	private var coinCross:FlxText;
+	private var score:FlxText;
+	private var time:FlxText;
+	private var coins:FlxText;
+	//This class is a static member of PlayState, and should be called from there.
+	
+	/**
+	 * 
+	 * @param	x	The X coordinate of the top left of the HUD
+	 * @param	y	The Y coordinate of the top left of the HUD
+	 * @param	playerName The name displayed above the score. Will be converted to upper case.
+	 */
+	public  function new(x:Int, y:Int, playerName:String) 
+	{
+		//creates a HUD
+		super();
+		//put the player name in the top left
+		scoreLabel = new FlxText(0 + x, 0 + y, 100, playerName.toUpperCase(), 16);
+		scoreLabel.setBorderStyle(FlxTextBorderStyle.SHADOW, FlxColor.BLACK, 3);
+		//put the score below the player name
+		score = new FlxText(0 + x, 16 + y, 100, "00000000", 16);
+		score.setBorderStyle(FlxTextBorderStyle.SHADOW, FlxColor.BLACK, 3);
+		//put the time label in the top right
+		timeLabel = new FlxText(580 + x, 0 + y, 100, "TIME", 16);
+		timeLabel.setBorderStyle(FlxTextBorderStyle.SHADOW, FlxColor.BLACK, 3);
+		//put the time below the time label
+		time = new FlxText(597 + x, 16 + y, 100, "000", 16);
+		time.setBorderStyle(FlxTextBorderStyle.SHADOW, FlxColor.BLACK, 3);
+		//the "X" in the coin counter
+		coinCross = new FlxText(300 + x, 0 + y, 20, "X", 16);
+		coinCross.setBorderStyle(FlxTextBorderStyle.SHADOW, FlxColor.BLACK, 3);
+		//number of coins to the right of the X
+		coins = new FlxText (316 + x, 0 + y , 50, "00", 16);
+		//the coin icon
+		coins.setBorderStyle(FlxTextBorderStyle.SHADOW, FlxColor.BLACK, 3);
+		coinIcon = new FlxSprite(285 + x, 3 + y, AssetPaths.coinIcon__png);
+		
+		//add everything to the group.
+		add(scoreLabel);
+		add(score);
+		add(timeLabel);
+		add(time);
+		add(coinCross);
+		add(coins);
+		add(coinIcon);
+	}
+	
+	/**
+	 * 
+	 * @param	newTime The time you want displayed on the HUD. Displays last 3 digits of number.
+	 */
+	private function handleTimeUpdate(newTime:Int):Void{
+		//trim number down to last 3 digits if over 3 digits
+		var parsedTime:String = Std.string(newTime);
+		var timeString:String = "000";
+		var parsedLength:Int = parsedTime.length;
+		timeString = timeString.substr(0, -parsedLength) + parsedTime;
```

Comment:
I noticed that in this function and in 'handleCoinsUpdate' you have a line to set a their string values, but in the if statement following that you either set that string to that same value again or another format. Seems redundant, but there may have been a reason you needed to do it this way.